spec/duration.html

```diff
@@ -1095,16 +1102,16 @@ <h1>
         1. If ! IsValidDuration(_years_, _months_, _weeks_, _days_, _hours_, _minutes_, _seconds_, _milliseconds_, _microseconds_, _nanoseconds_) is *false*, throw a *RangeError* exception.
         1. If _newTarget_ is not present, set _newTarget_ to %Temporal.Duration%.
         1. Let _object_ be ? OrdinaryCreateFromConstructor(_newTarget_, *"%Temporal.Duration.prototype%"*, « [[InitializedTemporalDuration]], [[Years]], [[Months]], [[Weeks]], [[Days]], [[Hours]], [[Minutes]], [[Seconds]], [[Milliseconds]], [[Microseconds]], [[Nanoseconds]] »).
-        1. Set _object_.[[Years]] to _years_.
-        1. Set _object_.[[Months]] to _months_.
-        1. Set _object_.[[Weeks]] to _weeks_.
-        1. Set _object_.[[Days]] to _days_.
-        1. Set _object_.[[Hours]] to _hours_.
-        1. Set _object_.[[Minutes]] to _minutes_.
-        1. Set _object_.[[Seconds]] to _seconds_.
-        1. Set _object_.[[Milliseconds]] to _milliseconds_.
-        1. Set _object_.[[Microseconds]] to _microseconds_.
-        1. Set _object_.[[Nanoseconds]] to _nanoseconds_.
+        1. Set _object_.[[Years]] to ℝ(𝔽(_years_)).
```

Comment:
if IsValidDuration is true, shouldn’t these values already be float64-representable?

Reply:
Not necessarily, an integer > MAX_SAFE_INTEGER and ≤ MAX_VALUE shouldn't make IsValidDuration return false, but ℝ(𝔽()) converts it to float64-representable.